hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -914,6 +950,97 @@ private void validateAllColumnStats(
     validate(metadataBasedColStats, fsBasedColStats, partitionPath, "column stats");
   }
 
+  private void validatePartitionStats(
+      HoodieMetadataValidationContext metadataTableBasedContext,
+      Set<String> baseDataFilesForCleaning,
+      List<String> allPartitions) throws Exception {
+
+    HoodieSparkEngineContext engineContext = new HoodieSparkEngineContext(jsc);
+    HoodieData<HoodieMetadataColumnStats> partitionStatsUsingColStats = getPartitionStatsUsingColStats(metadataTableBasedContext,
+        baseDataFilesForCleaning, allPartitions, engineContext);
+
+    TableSchemaResolver schemaResolver = new TableSchemaResolver(metaClient);
```

Comment:
Please check if we need to extract the schema resolver as a member variable. I see it's getting instantiated at multiple places in the class.

Reply:
Addressed